src/main/java/com/ace/alfox/controllers/ActionController.java

```diff
@@ -3,32 +3,42 @@
 import com.ace.alfox.game.models.Player;
 import com.ace.alfox.lib.ActionFactory;
 import com.ace.alfox.lib.ActionResult;
-import com.ace.alfox.lib.PlayerFactory;
-import org.springframework.http.*;
+import com.ace.alfox.lib.Database;
+import org.dizitart.no2.NitriteId;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpSession;
-
-import java.io.*;
 import java.util.Map;
 
 @RestController
 class ActionController {
 
+    @Autowired
+    private Database db;
+    @Autowired
+    private ActionFactory actionFactory;
+
     @RequestMapping(value = "/perform/{command}", method = { RequestMethod.GET, RequestMethod.POST })
-    public ResponseEntity home(HttpServletRequest request, @PathVariable String command, @RequestBody Map<String, Object> params) throws IOException {
+    public ResponseEntity home(HttpServletRequest request, @PathVariable String command, @RequestBody Map<String, Object> params) {
         HttpSession session = request.getSession(true);
-        String playerId = (String) session.getAttribute("cid");
-
-        Player player = PlayerFactory.fetch(playerId);
-        if(player == null) {
-            return ResponseEntity.notFound().build();
+        NitriteId playerId = (NitriteId) session.getAttribute("pid");
+
+        if(playerId == null) {
+            //return ResponseEntity.notFound().build();
+            // Instead of just returning 404, create a test player
+            Player _player = new Player();
+            db.players.insert(_player);
+            session.setAttribute("pid", _player.id);
+            playerId = _player.id;
         }
 
-        ActionFactory doSomething = ActionFactory.named(command).withParameters(params);
+        Player player = db.players.getById(playerId);
+        ActionFactory doSomething = actionFactory.named(command).withParameters(params);
```

Comment:
Why is the ActionFactory returning an ActionFactory. Almost seems like it's an ActionFactory Factory?

Seems intuitive to me to rename `withParameters` to `with` and then rename `p` to `params`. So then you end up with:
```java
    public ActionFactory with(Map<String, Object> params) {
        return new ActionFactory(action, params);
    }

    actionFactory.named(command).with(params);
```

Any reason to not combine this?

```java
actionFactory.get(command, params)
```

Where params is optional? I haven't super gone through the code yet so maybe it will be more clear once I do that.

Reply:
I totally agree. An artifact of the past.